### ISSUE TITLE
string `dtypes` not being observed when using `Frame.from_tsv` on integer-like strings

### ISSUE THREAD

**Issue by linclelinkpart5** [static-frame/static-frame#130]
`sf_test.tsv` contents:
```
a	b
001	009
010	090
100	900
```
Usage and unexpected behavior:
```
In [2]: sf.Frame.from_tsv('sf_test.tsv', dtypes={'a': str, 'b': str})                                                                                                    
Out[2]: 
<Frame>
<Index> a      b      <<U1>
<Index>
0       1      9
1       10     90
2       100    900
<int64> <<U21> <<U21>
```
Note how the numbers are converted to string, but only after they seem to be converted to ints (leading zeroes dropped).

**Comment by flexatone** [comment]
Thanks for isolating this issue.

This derives from our usage of `np.genfromtxt`, which does not permit specifying `dtype` per column, and apparently interprets columns with a leading zero as integers. SF applies passed `dtypes` after `np.genfromtxt` realization, and thus cannot intervene in the interpretation of these values first as integers that we then convert to strings.

Replacing our usage of a `np.genfromtxt` is a mid-term goal, but not likely in the 0.5 series, so for now please find a workaround.

**Comment by flexatone** [comment]
This will be corrected with #651

**Comment by flexatone** [comment]
Confirming that this is corrected with the new implementation.

With default type parsing, the zeros are disregarded and integers are returned; however, if `str` is specifically requested, we retain and round-trip the zero padded strings.

```python
>>> f = sf.Frame.from_dict({'A': ('0003','0001'), 'B': ('090', '100')})                                                                                                                                    
>>> f                                                                                                                                                                                                      
<Frame>
<Index> A     B     <<U1>
<Index>
0       0003  090
1       0001  100
<int64> <<U4> <<U3>
>>> f.to_tsv('/tmp/test.tsv')                                                                                                                                                                              

>>> sf.Frame.from_tsv('/tmp/test.tsv', index_depth=1)                                                                                                                                                      
<Frame>
<Index> A       B       <<U1>
<Index>
0       3       90
1       1       100
<int64> <int64> <int64>

>>> sf.Frame.from_tsv('/tmp/test.tsv', index_depth=1, dtypes=str)                                                                                                                                          
<Frame>
<Index> A     B     <<U1>
<Index>
0       0003  090
1       0001  100
<<U1>   <<U4> <<U3>
```
